src/goose.rs

```diff
@@ -1222,6 +1204,27 @@ impl GooseClient {
                     }
                     raw_request.set_status_code(Some(status_code));
                     raw_request.set_final_url(r.url().as_str());
+
+                    // Load test client was redirected.
+                    if raw_request.url != raw_request.final_url {
+                        let base_url = self.base_url.read().await.to_string();
+                        // Check if the URL redirected started with the load test base_url.
+                        if !raw_request.final_url.starts_with(&base_url) {
+                            let redirected_base_url = match Url::parse(&raw_request.final_url) {
+                                // Use URL to grab base_url, which is everything up to the path.
+                                Ok(base_url) => base_url[..url::Position::BeforePath].to_string(),
```

Comment:
@LionsAd I'm looking for the most "correct" way to extract the `base_url` from an arbitrary URL, with the least possible assumptions. The [`url`](https://docs.rs/url) module seems to have an internal concept of a base_url, but it's not exposed directly as far as I can see. My solution (which works in my testing so far) is to convert the path to a `url::Url`, convert that to a Rust slice, and then copy everything from the start of the slice to `url::Position::BeforePath` to a string.

See https://docs.rs/url/*/url/enum.Position.html for the relevant docs.

Reply:
I think that sounds good to me - unless there is a function to extract a base url, I am not sure it makes sense.